packages/plugin-npm/sources/NpmSemverResolver.ts

```diff
@@ -54,6 +54,7 @@ export class NpmSemverResolver implements Resolver {
     });
 
     const candidates = Object.keys(registryData.versions)
+      .filter(version => !!semverUtils.valid(version))
```

Comment:
This will loop over the array three times and parse the version twice, could you refactor this to use `miscUtils.mapAndFilter` instead so it's one iteration and one parse?

Reply:
done